### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by forivall**
Basically, i'm dealing with a google drive with lots of folders, only some of which contain documents. Paging meant that all of the folders werent being traversed, and the many folders meant that traversing it was slooow. By checking multiple folders in one request, i'm able to greatly speed up the traversal.

Also, some minor refactoring to use async/await, typescript definitions for the options, jsdoc typing (mostly to help myself figure out what was going on), [~a minor bugfix in converters~](https://github.com/cedricdelpoux/gatsby-source-google-docs/commit/ab8eeea69b498df2521eb7a78606fc4a0fefd6c8#r42979090), and a spelling fix in docs.

### CODE REVIEW

**Comment by forivall** [comment]
(also, the branch name is called cache because i was originally going to cache the requests, but figured out a better way to fetch the files)

**Comment by cedricdelpoux** [comment]
hi @forivall, thank you for your work! 
Bad luck for you, I made some refactoring yesterday to add a lot of tests.

Could you rebase your branch on master to resolve conflicts? 

Thank you very much

**Review by cedricdelpoux — CHANGES_REQUESTED** [review]
Please resolve conflicts

**Comment by forivall** [comment]
Note that performance can shift a bit with different [`BATCH_SIZE`](https://github.com/forivall/gatsby-source-google-docs/blob/662fcab5/utils/google-drive.js#L125) settings. It could be made an option, as the optimum setting depends on the shape of a user's drive folders.

I also could do some refactoring into a class to manage the paging, as that would probably make things easier to read and reason about with the control flow.

File: utils/google-drive.js [hunk 1]
```diff
@@ -1,13 +1,36 @@
 const {google} = require("googleapis")
 const _kebabCase = require("lodash/kebabCase")
+const _chunk = require("lodash/chunk")
+const _flatten = require("lodash/flatten")
 const GoogleOAuth2 = require("google-oauth2-env-vars")
 const yamljs = require("yamljs")
+const wyt = require("@forivall/wyt")
```
Comment (cedricdelpoux) [inline utils/google-drive.js:7]: Please mention me or update this when your fork will be merged

File: utils/google-drive.js [hunk 2]
```diff
@@ -59,104 +82,205 @@ const updateMetadata = ({metadata, fieldsDefault = {}, fieldsMapper = {}}) => {
...
+ */
+
+/**
+ * @typedef FetchDocumentsOptions
+ * @property {import('googleapis').drive_v3.Drive} drive
+ * @property {DocumentFetchParent[]} parents
+ */
+
+// 10 per 1.5 seconds.
+const rateLimit = wyt(10, 1500)
```
Comment (cedricdelpoux) [inline utils/google-drive.js:108]: Did you find an official documentation about this or did you lower until it works?

**Review by cedricdelpoux — APPROVED** [review]

**Comment by cedricdelpoux** [comment]
Release in `2.0.0-beta.26` version

File: utils/google-drive.js (hunk 1 above)
Comment (forivall) [reply utils/google-drive.js:7]: Will do! it's https://github.com/maxkueng/wyt/pull/6 fyi

**Pull request merged**